### ISSUE TITLE
duplicate DOI makes DataCite export fail

### ISSUE THREAD

**Issue by ppanero** [inveniosoftware/invenio-rdm-records#811]
Package version (if known): v0.33.15

## Describe the bug
If the record is created with a DOI, and the same DOI gets added as alternate identifier. The DataCite export will add the same `{identifier, type}` pair to the list of identifiers and it will fail.

```
{
  "pids": {
    "doi": {
      "identifier": "10.81088/12345-abcde",
      "provider": "datacite",
      "client": "datacite"
    }
  },
  ...
  "metadata": {
    "identifiers": [
      {
        "identifier": "10.81088/12345-abcde",
        "scheme": "doi"
      }
    ],
  }
}
```

## Expected behavior

The serializer should be able to deal with duplicates.

## Additional context

Sentry error link: https://inveniordm-sentry.web.cern.ch/share/issue/6b70f6d5cef74c3cbb3139775e680340/

**Comment by tmorrell** [comment]
This error is coming from the datacite library, because DataCite JSON doesn't support DOIs as alternate identifiers. Technically alternate identifiers should refer to exactly the same instance (location and file), and there shouldn't be a second DOI to the exact same object. If you send JSON with alternative identifier DOIs to DataCite, they are removed. 

I think the best solution is to remove 'DOI' from the schema options for alternate identifiers, and tell people to use related identifiers instead. DataCite doesn't define schema types for alternate identifiers, so we have full control of what types are allowed.

If you want to instead change the error behavior of datacite, let me know and I can put something together.

**Comment by ppanero** [comment]
Hi @tmorrell thanks for looking into this. As for removing it from the list, it might not fully suit a wider case (@lnielsen can elaborate more on that). However, the is an issue with the identifier(s) field:

I think the DataCite is wrong... The [DataCite definition](https://schema.datacite.org/meta/kernel-4.3/doc/DataCite-MetadataKernel_v4.3.pdf) mentions two fields: *identifier* and *alternateIdentifier* (being the first the DOI). 
![Screenshot 2021-09-03 at 10 00 38](https://user-images.githubusercontent.com/6756943/131972932-305ee330-8db7-43b4-bc48-a9252393a0e3.png)
![Screenshot 2021-09-03 at 10 00 30](https://user-images.githubusercontent.com/6756943/131972936-8572cfee-0cb7-4e06-8f5e-40b79cbeb41c.png)

However, we have only "identifiers" (See [serializer](https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/resources/serializers/datacite/schema.py#L191). Probably because the datacite library might also be wrong (?) in the [tests](https://github.com/inveniosoftware/datacite/blob/master/tests/data/datacite-v4.3-full-example.json#L88)

My idea is:
- add identifier field for the DOI
- any other identifier should go to alt/related identifiers

WDYT?

**Comment by lnielsen** [comment]
The DataCite JSON merged both fields into ``identifiers`` hence the difference between XML/Schema and JSON.

**Comment by lnielsen** [comment]
RDM will need to allow DOIs in alternate identifiers in the data model, because an RDM instance is not required to have DOI support (and thus there wouldn't be a way to record the DOI for these instances). However, perhaps we just need to remove DOIs in alternate identifiers from the DataCite serialization format.

**Comment by tmorrell** [comment]
Fixing the serialization to remove alternative identifier DOIs makes sense to me, since DataCite won't take them anyway. If no one gets to it earlier, I'll put together a PR next week.

**Comment by tmorrell** [comment]
This is still an issue, will be resolved by #829
